### PR TITLE
feat: push expired tiles into job queue and major refactoring

### LINKED ISSUE #2: insert expired tiles to job queue
`append` command outputs expired tiles in web mercator projection. parse those values into bboxes and push into a rendering queue.
make this feature configurable as well as the upload of the expired tiles to s3, so they both can be activated in parallel

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                       |
| Breaking change | ✔                                                                        |
| Deprecations    | ✔                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Closes #2

### CODE REVIEW

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/commandRunner/OsmCommandRunner.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { Logger } from '@map-colonies/js-logger';
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/commandRunner/OsmCommandRunner.ts]: done.

File: README.md [hunk 2]
```diff
@@ -60,7 +60,12 @@ After the initial creation of a project, data can be appended.
 A project can be divided into multiple sub parts, each sub part's lua script needs to be placed in the bucket under the projectId meaning with the projectId as the script's key prefix.
 
 On append a config representing the job is required, the config specifies each sub part's s3 lua script key and it's expired tiles zoom levels.
-The expired tiles of each sub part will be uploaded to the bucket as well under `/projectId/subId/sequenceNumber/expire.list`.
+
+The output of the append command of `osm2pgsql` is an expire list of tiles in requested zoom levels.
+
+The expired tiles list can be uploaded to a source of your choice, `s3` or `queue`.
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply README.md]: done.

File: src/index.ts [hunk 3]
```diff
@@ -2,20 +2,37 @@
...
-import { ExitCodes, EXIT_CODE, ON_SIGNAL } from './common/constants';
-import { getCli } from './cli';
+import { ExitCodes, EXIT_CODE, SERVICES } from './common/constants';
+import { getCli } from './cli/cli';
+import { ShutdownHandler } from './common/shutdownHandler';
+
+interface IError {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  error: (...data: any[]) => void;
+}
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/index.ts]: done.

File: src/containerConfig.ts [hunk 4]
```diff
@@ -3,43 +3,61 @@ import { logMethod } from '@map-colonies/telemetry';
...
+export const registerExternalValues = async (options?: RegisterOptions): Promise<DependencyContainer> => {
+  const shutdownHandler = new ShutdownHandler();
+
+  try {
+    const loggerConfig = config.get<LoggerOptions>('telemetry.logger');
+    // @ts-expect-error the signature is wrong
+    const logger = jsLogger({ ...loggerConfig, hooks: { logMethod } });
+
+    const httpClientConfig = config.get<object>('httpClient');
+    const httpClient = httpClientFactory(httpClientConfig);
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/containerConfig.ts]: done.

File: src/containerConfig.ts [hunk 5]
```diff
@@ -3,43 +3,61 @@ import { logMethod } from '@map-colonies/telemetry';
...
+      { token: CREATE_MANAGER_FACTORY, provider: { useFactory: createManagerFactory } },
+      { token: APPEND_COMMAND_FACTORY, provider: { useFactory: appendCommandFactory } },
+      { token: APPEND_MANAGER_FACTORY, provider: { useFactory: appendManagerFactory } },
+      { token: SERVICES.CONFIG, provider: { useValue: config } },
+      { token: SERVICES.LOGGER, provider: { useValue: logger } },
+      { token: SERVICES.TRACER, provider: { useValue: tracer } },
+      { token: SERVICES.HTTP_CLIENT, provider: { useValue: httpClient } },
+      { token: SERVICES.CONFIG_STORE, provider: { useValue: configStore } },
+      { token: EXIT_CODE, provider: { useValue: ExitCodes.SUCCESS } },
+    ];
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/containerConfig.ts]: done.

File: helm/README.md [hunk 6]
```diff
@@ -10,6 +10,8 @@
 
 **postgres:**
 
+the postgres for osm2pgsql
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply helm/README.md]: done.

File: .gitignore [hunk 7]
```diff
@@ -112,3 +112,5 @@ dist
 # Jest
 jest_html_reporters.html
 reports
+
+local.json
```
Comment (CptSchnitz) [inline .gitignore:116]: mb local*.json ?
and new line is missing.

File: helm/README.md [hunk 8]
```diff
@@ -40,11 +59,19 @@
 **cli:**
 
 - `cli.command` - the command to be run create or append
+
+*append*
 - `cli.append.replicationUrl` - the source of replication
-- `cli.append.limit.value` - should limit the amount of appends in a single run
+- `cli.append.limit.enabled` - should limit the amount of appends in a single run
```
Comment (CptSchnitz) [inline helm/README.md:65]: should? use a more decisive wording.

File: package.json [hunk 9]
```diff
@@ -39,21 +39,25 @@
     "@map-colonies/js-logger": "^0.0.3",
     "@map-colonies/read-pkg": "0.0.1",
     "@map-colonies/telemetry": "3.0.0",
+    "@map-colonies/tile-calc": "^0.1.3",
+    "@mapbox/sphericalmercator": "^1.2.0",
     "@opentelemetry/api": "1.0.1",
     "@opentelemetry/api-metrics": "0.23.0",
     "@opentelemetry/instrumentation-http": "0.23.0",
+    "@types/mapbox__sphericalmercator": "^1.1.5",
```
Comment (CptSchnitz) [inline package.json:47]: move to dev dep

File: src/cli/checks/index.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        throw new Error(`${errorPrefix}, ${dumpSource} does not exist locally`);
+      }
+    } else if (isWebUri(dumpSource) === undefined) {
+      throw new Error(`${errorPrefix}, ${dumpSource} is not a valid web uri`);
+    }
+
+    return true;
+  };
+  return check;
+};
```
Comment (CptSchnitz) [inline src/cli/checks/index.ts:71]: some documentation on the functions would be nice.

File: src/validation/validator.ts [hunk 11]
```diff
@@ -1,35 +1,24 @@
...
   content?: T;
 }
 
-@injectable()
-export class Validator<T> {
-  private readonly ajv: Ajv;
-  public constructor(@inject(SERVICES.LOGGER) private readonly logger: Logger) {
-    this.ajv = new Ajv({ $data: true });
-    ajvKeywords.default(this.ajv);
+export function validateBySchema<T>(content: unknown, schema: JSONSchemaType<T>): ValidationResponse<T> {
```
Comment (CptSchnitz) [inline src/validation/validator.ts:15]: its more of an ajv wrapper, than validate by schema.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/validation/validator.ts (hunk 11 above)
Comment (melancholiai) [reply src/validation/validator.ts]: done.

File: src/cli/checks/index.ts (hunk 10 above)
Comment (melancholiai) [reply src/cli/checks/index.ts]: done.

File: package.json (hunk 9 above)
Comment (melancholiai) [reply package.json]: done.

File: helm/README.md (hunk 8 above)
Comment (melancholiai) [reply helm/README.md]: done.

File: .gitignore (hunk 7 above)
Comment (melancholiai) [reply .gitignore]: done.

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**